File: LeetCode/3005.CountElements_With_Maximum_Frequency/susan/3005.CountElements_With_Maximum_Frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        int max = 0;
        int res = 0;
        Map<Integer, Integer> map = new HashMap<>();
        
        for(int num : nums){
            map.put(num,map.getOrDefault(num,0)+1);
        }
        
        for(int i : map.keySet()){
            max = Math.max(max, map.get(i));
        }
        
        for(int i : map.keySet()){
            if(map.get(i) == max){
                res +=map.get(i);
            }
        }
        
        
        return res;
        
    }
}

File: LeetCode/3005.Count_Elements_With_Maximum_Frequency/3005.Count_Elements_With_Maximum_Frequency.swift
class Solution {
    func maxFrequencyElements(_ nums: [Int]) -> Int {
        var buckets = Array(repeating: 0, count: 101)

        var maxFreq = 0
        for num in nums {
            buckets[num] += 1
            maxFreq = max(maxFreq, buckets[num])
        }

        return buckets.reduce(0) { pre, cur in
            pre + (cur == maxFreq ? cur : 0)
        }
    }
}
